### ISSUE TITLE
[BUG] FBD analyses w/ FossilTipTimeSlideUniformProposal hanging after restart

### ISSUE THREAD

**Issue by davidcerny** [revbayes/revbayes#516]
**Describe the bug**
When the `mvFossilTimeSlideUniform` move is applied to all fossils simultaneously, the initial analysis will run fine, but will hang indefinitely if an attempt is made to restart it from a checkpoint file.

**To Reproduce**
This chunk of Rev code from the `model_FBDP.Rev` script in the [combined-evidence FBD tutorial](https://revbayes.github.io/tutorials/fbd/fbd_specimen.html) is the culprit:

```
for(i in 1:fossils.size())
{
    t[i] := tmrca(fbd_tree, clade(fossils[i]))

    a_i = fossils[i].getMinAge()
    b_i = fossils[i].getMaxAge()

    F[i] ~ dnUniform(t[i] - b_i, t[i] - a_i)
    F[i].clamp( 0 )
}

# Add a move to sample the fossil times #
moves.append( mvFossilTimeSlideUniform(fbd_tree, origin_time, weight=5.0) )
```

Interestingly, if we place the move inside the loop and only apply it to one fossil at a time, like this:

```
for(i in 1:fossils.size())
{
    t[i] := tmrca(fbd_tree, clade(fossils[i]))

    a_i = fossils[i].getMinAge()
    b_i = fossils[i].getMaxAge()

    F[i] ~ dnUniform(t[i] - b_i, t[i] - a_i)
    F[i].clamp( 0 )

    # Add a move to sample the fossil times #
    moves.append( mvFossilTimeSlideUniform(fbd_tree, origin_time, fossils[i], weight=5.0) )
}
```

the restarted analysis will run just fine. Similarly, if we replace `mvFossilTimeSlideUniform` by the closely related `mvFossilTipTimeUniform`, there are no checkpointing issues, regardless of whether the move is applied to a single fossil (inside the loop) or all fossils simultaneously (outside the loop).

**Expected behavior**
We should be able to restart the analysis from a checkpoint file regardless of whether `mvFossilTimeSlideUniform` is applied to a single fossil or all fossils simultaneously.

**Computer info**
macOS 14.5 (Sonoma), ARM64 architecture, RevBayes 1.2.5-preview, latest version of the development branch (ac87f90).

**Comment by bredelings** [comment]
Can you figure out what it is doing while it is hanging?  If you compile a debug version, run it inside gdb, and do CTRL-C, you should be able to see what loop it is getting stuck in.

**Comment by bredelings** [comment]
I'm not quite sure how to reproduce this...

**Comment by davidcerny** [comment]
> Can you figure out what it is doing while it is hanging?

This is the output I got from lldb:

```
Iter        |      Posterior   |     Likelihood   |          Prior   |     age_extant   |   num_samp_anc   |    origin_time   |    elapsed   |        ETA   |
-------------------------------------------------------------------------------------------------------------------------------------------------------------
Process 96802 stopped
* thread #1, queue = 'com.apple.main-thread', stop reason = signal SIGSTOP
    frame #0: 0x000000010111d1d0 rb`RevBayesCore::FossilTipTimeSlideUniformProposal::doProposal(this=0x0000600002f72be0) at FossilTipTimeSlideUniformProposal.cpp:241:23
   238 	        {
   239 	            new_age = 2.0 * max_age - new_age;
   240 	        }
-> 241 	    } while ( new_age < min_age || new_age > max_age );
   242 	    
   243 	    // set the age
   244 	    node->setAge( new_age );
Target 0: (rb) stopped.
```

> I'm not quite sure how to reproduce this...

This worked for me:

1. Go to the linked tutorial;
2. Download the files;
3. Comment out the line that reads `mymcmc.initializeFromCheckpoint("output/state.state")` in the master script (`mcmc_CEFBDP_Specimens.Rev`) to run the analysis for the first time;
4. Try restarting the analysis by uncommenting the line in question and sourcing the master script again.

To make it more convenient, I'm also attaching the files here:

[fbd_checkpoint.zip](https://github.com/user-attachments/files/16344939/fbd_checkpoint.zip)

**Comment by bjoelle** [comment]
yes, this is where it hangs in my trials as well, and that also explains why it's not happening in `mvFossilTipTimeUniform` I would expect since it's not doing that 
but same as Ben I don't understand why the change in pointers would affect this

**Comment by bjoelle** [comment]
see PR #517 for proposed fix
